libclamav/cache.c

```diff
@@ -31,30 +31,21 @@
 #include "mpool.h"
 #include "clamav.h"
 #include "cache.h"
+#include "math.h"
 #include "fmap.h"
 
 #include "clamav_rust.h"
 
-/* The number of root trees and the chooser function
+/* The chooser function
    Each tree is protected by a mutex against concurrent access */
-/* #define TREES 1 */
-/* static inline unsigned int getkey(uint8_t *hash) { return 0; } */
-#define TREES 256
-static inline unsigned int getkey(uint8_t *hash)
+static inline unsigned int getkey(uint8_t *hash, size_t trees)
 {
     if (hash) {
-        return *hash;
+        return (hash[0] | (((unsigned int)hash[1]) << 8)) % trees;
```

Comment:
> It took me embarrassingly long to understand this line of code.

Same here. I actually didn't write it so I can't take full credit for it... it originated in commit https://github.com/Cisco-Talos/clamav/commit/438f0550d6d61a7ad308c9f56eb2abf1f453523f

> I like it, but it could use a comment to explain it so others' eyes don't glaze over when reading it as mine would.

I've added a comment - if you think it can improved, please let me know :)

Reply:
Your comment is great! Thanks!